### PR TITLE
Add RebootRoleInstanceAsync to allow programmatic rebooting of cloud service instance

### PR DESCRIPTION
**Pull request by joelverhagen**
Progress on https://github.com/NuGet/Engineering/issues/1527.

Make long lines more readable and used string interpolation instead of `string.Format`.

### CODE REVIEW

File: src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs [hunk 1]
```diff
@@ -41,7 +41,14 @@ public AzureManagementAPIWrapper(IAzureManagementAPIWrapperConfiguration configu
             _clientCredential = new ClientCredential(configuration.ClientId, configuration.ClientSecret);
         }
 
-        public async Task<string> GetCloudServicePropertiesAsync(string subscription, string resourceGroup, string name, string slot, CancellationToken token)
+        public async Task RebootRoleInstanceAsync(
```
Comment (loic-sharma) [inline src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs]: >RebootRoleInstanceAsync [](start = 26, length = 23)

This API is cloud service specific, right? Should "CloudService" be in this method name? We may want to add rebooting of App Services later

Comment (joelverhagen) [reply src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs]: Sure, good idea.

Comment (joelverhagen) [reply src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs]: Fixed.

File: src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs [hunk 2]
```diff
@@ -63,14 +70,71 @@ public AzureManagementAPIWrapper(IAzureManagementAPIWrapperConfiguration configu
...
+            }
 
-            string requestUrl = string.Format(RequestUrlFormat, subscription, resourceGroup, name, slot);
+            if (string.IsNullOrEmpty(roleInstance))
+            {
+                throw new ArgumentException(nameof(roleInstance));
+            }
 
-            return await MakeAzureRequest(requestUrl, token);
+            var requestUrl = "https://management.azure.com" +
```
Comment (agr) [inline src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs]: Hostname should at least be a constant. Ideally it should be passed from outside (to a constructor?).

**Pull request merged**